docs/source/pipelines.rst

```diff
@@ -4,194 +4,321 @@
+        git clone https://github.com/mlflow/mlp-regression-template
+        # Enter the MLflow Pipelines Regression Template directory and configure dbx within it
+        cd mlp-regression-template
+        dbx configure
+        # Use dbx to enable syncing from the repository directory to Databricks Repos
+        dbx sync repo -d mlp-regression-template
+        # Iteratively make changes to files in the repository directory and observe that they
+        # are automatically synced to Databricks Repos
+        ...
+
+4. Test changes by running the pipeline and observing the results it produces. MLflow Pipelines
+   intelligently caches results from each :ref:`Pipeline Step <steps-key-concept>`, ensuring that
+   steps are only executed if their inputs, code, or configurations have changed, or if such
+   changes have occurred in dependent steps. Once you are satisfied with the results of
+   your changes, commit them to a branch of the :ref:`Pipeline Repository
+   <pipeline-templates-key-concept>` in order to ensure reproducibility, and share or review the
+   changes with your team.
 
-Pipeline Repositories
-#####################
+    .. note::
+      Before testing changes in a staging or production environment, it is recommended that you
+      commit the changes to a branch of the
+      :ref:`Pipeline Repository <pipeline-templates-key-concept>` to ensure reproducibility.
+
+    .. note::
+      By default, MLflow Pipelines caches results from each :ref:`Pipeline Step
+      <steps-key-concept>` within the ``.mlflow`` subdirectory of the home folder on the
+      local filesystem. The ``MLFLOW_PIPELINES_EXECUTION_DIRECTORY`` environment variable can
+      be used to specify an alternative location for caching results.
+
+.. _pipeline-templates:
+
+Pipeline Templates
+------------------
+
+MLflow Pipelines currently offers the following predefined templates that can be easily customized
+to develop and deploy high-quality, production-ready models for your use cases:
+
+.. _regression-template:
+
+- **MLflow Pipelines Regression Template**: The `MLflow Pipelines Regression Template
+  <https://github.com/mlflow/mlp-regression-template>`_ is designed for developing and scoring
+  regression models. For more information, see the |Regression Template reference guide| and
+  the :ref:`Regression Pipeline API documentation <mlflow-regression-pipeline>`.
+
+Additional pipelines for a variety of ML problems and MLOps tasks are under active development.
+
+
+Detailed reference guide
+------------------------
+
+.. _pipeline-template-structure:
+
+Template structure
+~~~~~~~~~~~~~~~~~~
 
-Pipelines are structured as git repositories with a standardized directory hierarchy. The following
-example provides an overview of the pipeline repository structure.  It is adapted from the
-|MLflow Regression Pipeline repository|.
+Pipeline Templates are git repositories with a standardized, modular layout. The following
```

Comment:
What does modular mean here? The pipeline is modularized but I'm not sure how that term applies to the bare file structure.

Reply:
Good question! It means that the pipeline definition is decomposed into individual steps that each have their own code confs and code files, in contrast to a notebook that globs together all of the pipeline content in the same place and is hard to review / productionize.